### PR TITLE
Bug/CXINT-2438: Update Sort Keys in Spartacus PDF Invoices Component 

### PR DESCRIPTION
**Pull request by anjana-bl**
https://jira.tools.sap/browse/CXINT-2438

### CODE REVIEW

File: feature-libs/pdf-invoices/components/invoices-list/invoices-list.component.ts [hunk 1]
```diff
@@ -51,8 +51,8 @@ export class InvoicesListComponent implements OnInit, OnDestroy {
   sort = 'byInvoiceIdAsc';
 
   protected sortMapping: { [key: string]: string } = {
-    byCreatedAtAsc: 'invoiceDate:asc', //TODO: (CXINT-2438) update the sort code after the API is changed
-    byCreatedAtDesc: 'invoiceDate:desc', //TODO: (CXINT-2438) update the sort code after the API is changed
+    byCreatedAtAsc: 'createdAt:asc',
+    byCreatedAtDesc: 'createdAt:desc',
```
Comment (Platonn) [inline feature-libs/pdf-invoices/components/invoices-list/invoices-list.component.ts]: Could you help me understand the backward compatibility consequences of this change?

I can understand that users of the newest backend API and newest Spartacus will get desired behavior.
And users of the old backend API and old Spartacus will also get desired correct behavior.

But does this change mean that some customers that use the old version of the backend API, but the newest Spartacus will encounter problems?

Analogical question: for users of new API but old Spartacus - will they encounter problems?

Should we wrap this change with an opt-in feature flag?

Comment (anjana-bl) [reply feature-libs/pdf-invoices/components/invoices-list/invoices-list.component.ts]: yes it should be wrapped in opt-in feature flag, i have made the necessary changes

File: feature-libs/pdf-invoices/components/pdf-invoices-components.module.ts [hunk 2]
```diff
@@ -45,6 +45,9 @@ import { InvoicesListComponent } from './invoices-list/invoices-list.component';
           guards: [AuthGuard],
         },
       },
+      features: {
+        useUpdatedSortMappingForInvoices: false,
```
Comment (Platonn) [inline feature-libs/pdf-invoices/components/pdf-invoices-components.module.ts]: I recommend starting with the name of the library, e.g. `pdfInvoices` and if possible describe semantically the change: e.g. `sortByInvoiceDate`.
What do you think about: `pdfInvoicesSortByInvoiceDate`?

Comment (anjana-bl) [reply feature-libs/pdf-invoices/components/pdf-invoices-components.module.ts]: yes, it is better. have changed the name

**Review by Platonn — APPROVED** [review]

**Pull request merged**